internal/dataplane/kongstate/credentials.go

```diff
@@ -158,7 +166,7 @@ func (c *KeyAuth) SanitizedCopy() *KeyAuth {
 			// Consumer field omitted
 			CreatedAt: c.CreatedAt,
 			ID:        c.ID,
-			Key:       redactedString,
+			Key:       randRedactedString(),
```

Comment:
Wouldn't it be slightly more robust w.r.t. unique ID generation to use dependency injection in `KeyAuth` instead and check that whenever a sanitized copy is to be returned?

This way we don't use global `uuid.SetRand` but rather a local rand seed (hardcoded in tests).

`randRedactedString()` would then use that rand source.

WDYT?

Reply:
Yeah, I think it makes sense. We could take that further and inject a UUID generator: https://github.com/Kong/kubernetes-ingress-controller/pull/5983